lib/src/license_detection/license.dart

```diff
@@ -0,0 +1,139 @@
+// Copyright (c) 2017, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:meta/meta.dart';
+import 'package:pana/src/license_detection/tokenizer.dart';
+
+@sealed
+class License {
+  /// Name of the license, is empty in case of unknown license.
+  final String licenseName;
+
+  /// Original text from the license file.
+  final String content;
+
+  /// Normalized [Token]s created from the original text.
+  final List<Token> tokens;
+
+  /// A map of tokens and their count.
+  final Map<String, int> occurences;
+
+  License._(this.content, this.tokens, this.occurences, this.licenseName);
+
+  factory License.parse(String licenseName, String content) {
+    final tokens = tokenize(content);
+    final table = generateFrequencyTable(tokens);
+    return License._(content, tokens, table, licenseName);
+  }
+}
+
+// Commented out as they are not yet needed in this stage of detection.
+// class Checksum {
+//   /// Text for which the hash value was generated.
+//   final String text;
+
+//   /// [Crc-32][1] checksum value generated for text.
+//   ///
+//   /// [1]: https://en.wikipedia.org/wiki/Cyclic_redundancy_check
+//   final int crc32;
+
+//   Checksum(this.text, this.crc32);
+// }
+
+/// Contains deatils regarding the results of corpus license match with unknwown text.
+@sealed
+class LicenseMatch {
+  /// Name of the license detected from unknown text.
+  final String name;
+
+  /// Tokens that were found in the unknown text that matched to tokens
+  /// in any of the corpus license.
+  final List<Token> tokens;
+
+  /// Confidence score of the detected license.
+  final double confidenceScore;
```

Comment:
Should we say between 0 and 1? or what is the range here?
```suggestion
  /// Confidence score of the detected license ranging from 0 to 1.
  ///
  /// A confidence score of 1, means that after text normalization we have a
  /// perfect match.
  final double confidence;
```

Should we consider just calling the variable for `confidence`? because it's our confidence in the detected license as a number between 0 and 1.